app/src/shared/contexts/ModalApi/index.jsx

```diff
@@ -53,6 +54,20 @@ function useModalContext(): ContextProps {
         })
     }, [modals, isMounted])
 
+    // close all modals on route change
+    useEffect(() => () => {
+        if (!isMounted()) { return }
+
+        setModals((prevModals) => {
+            Object.keys(prevModals).forEach((modalId) => {
+                if (prevModals[modalId].reject && typeof prevModals[modalId].reject === 'function') {
```

Comment:
`typeof prevModals[modalId].reject === 'function'` should be enough.

Reply:
Left this as is finally, doesn't do any harm and the other conditions are similar.